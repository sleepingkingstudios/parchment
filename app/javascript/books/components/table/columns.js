const columns = [
  {
    label: 'Title',
    prop: 'title',
  },
  {
    label: 'Publisher',
    prop: 'publisherName',
  },
  {
    label: ' ',
    prop: 'actions',
    value: ' ',
  },
];

export default columns;
