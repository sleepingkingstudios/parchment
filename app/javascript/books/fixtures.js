import { load } from '../fixtures';

/* eslint-disable-next-line import/prefer-default-export */
export const booksData = load('books');
