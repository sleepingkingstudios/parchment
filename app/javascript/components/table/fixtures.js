import { humanizeList } from '../../utils/array';

export const formatCrewCapacity = (item) => {
  const count = item.crewCapacity;

  if (count === 1) { return '1 Kerbal'; }

  return `${count} Kerbals`;
};

export const formatExperiments = (item) => {
  humanizeList(item.experiments, { default: 'None' });
};

export const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Crew Capacity',
    prop: 'crewCapacity',
    value: formatCrewCapacity,
  },
  {
    label: 'Experiments',
    prop: 'experiments',
  },
];

export const rocketsData = [
  {
    id: '83f5e9e8-92df-4203-bc2e-4381370c2186',
    name: 'Apprentice',
    crewCapacity: 1,
    experiments: [
      'Crew Report',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '4bd7454c-3b15-4047-8d57-dfa831dac58e',
    name: 'Sorcerer',
    crewCapacity: 2,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '2fead083-d9ea-4f04-949e-d8c00e800cbb',
    name: 'Warlock',
    crewCapacity: 3,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
      'Gravity Scan',
    ],
  },
];
