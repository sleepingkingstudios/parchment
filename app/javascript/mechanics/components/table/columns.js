const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Description',
    prop: 'shortDescription',
  },
  {
    label: ' ',
    prop: 'actions',
    value: null,
  },
];

export default columns;
