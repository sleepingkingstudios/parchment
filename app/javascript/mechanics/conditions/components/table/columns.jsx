import ConditionsTableActions from './actions';

const columns = [
  {
    label: 'Name',
    prop: 'name',
    width: 3,
  },
  {
    label: 'Description',
    prop: 'shortDescription',
    width: 6,
  },
  {
    label: false,
    prop: 'actions',
    value: ConditionsTableActions,
    width: 3,
  },
];

export default columns;
