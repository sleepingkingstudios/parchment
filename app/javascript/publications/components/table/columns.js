import PublicationsTableActions from './actions';

const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Publisher',
    prop: 'publisherName',
  },
  {
    label: 'Date',
    prop: 'publicationDate',
  },
  {
    label: 'Playtest',
    prop: 'playtest',
  },
  {
    label: ' ',
    prop: 'actions',
    value: PublicationsTableActions,
  },
];

export default columns;
