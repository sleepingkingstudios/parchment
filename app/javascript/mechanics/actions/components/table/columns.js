import ActionsTableActions from './actions';
import { columns as mechanicsColumns } from '../../../components/table';

const columns = mechanicsColumns.slice();

columns[columns.length - 1] = {
  label: ' ',
  prop: 'actions',
  value: ActionsTableActions,
};

export default columns;
