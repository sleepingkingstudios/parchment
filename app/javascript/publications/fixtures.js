import { load } from '../fixtures';

const isOfficial = (publication) => {
  const { playtest, publisherName } = publication;

  return !playtest && publisherName === 'Wizards of the Coast';
};

const rawData = load('publications');

/* eslint-disable-next-line import/prefer-default-export */
export const publicationsData = rawData
  .map(publication => Object.assign(
    {},
    publication,
    {
      official: isOfficial(publication),
    },
  ));
