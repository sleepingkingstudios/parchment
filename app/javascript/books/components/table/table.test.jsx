import React from 'react';
import { shallow } from 'enzyme';

import BooksTable from './table';
import columns from './columns';
import { booksData } from '../../fixtures';

describe('<BooksTable />', () => {
  const onDelete = jest.fn();
  const defaultProps = { onDelete };
  const emptyMessage = 'There are no books matching the criteria.';

  describe('with books: empty Array', () => {
    const props = { ...defaultProps, books: [] };

    it('should render the Table', () => {
      const rendered = shallow(<BooksTable {...props} />);
      const table = rendered.find('Table');

      expect(table).toExist();
      expect(table).toHaveProp('columns', columns);
      expect(table).toHaveProp('data', []);
      expect(table).toHaveProp('message', emptyMessage);
      expect(table).toHaveProp('cellProps', { onDelete });
    });

    it('should match the snapshot', () => {
      const rendered = shallow(<BooksTable {...props} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('with books: Array with many items', () => {
    const props = { ...defaultProps, books: booksData };

    it('should render the Table', () => {
      const rendered = shallow(<BooksTable {...props} />);
      const table = rendered.find('Table');

      expect(table).toExist();
      expect(table).toHaveProp('columns', columns);
      expect(table).toHaveProp('data', booksData);
      expect(table).toHaveProp('cellProps', { onDelete });
    });

    it('should match the snapshot', () => {
      const rendered = shallow(<BooksTable {...props} />);

      expect(rendered).toMatchSnapshot();
    });
  });
});
