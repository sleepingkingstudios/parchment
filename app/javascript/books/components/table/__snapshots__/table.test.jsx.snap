// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BooksTable /> with books: Array with many items should match the snapshot 1`] = `
<Table
  cellProps={
    Object {
      "onDelete": [MockFunction],
    }
  }
  columns={
    Array [
      Object {
        "label": "Title",
        "prop": "title",
      },
      Object {
        "label": "Publisher",
        "prop": "publisherName",
      },
      Object {
        "label": " ",
        "prop": "actions",
        "value": " ",
      },
    ]
  }
  data={
    Array [
      Object {
        "abbreviation": "sff",
        "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
        "playtest": true,
        "publicationDate": "2013-08-15",
        "publisherName": "Spelljammer Publishing",
        "slug": "spectres-flumph-fortress",
        "title": "Spectres of Flumph Fortress",
      },
      Object {
        "abbreviation": "cif",
        "id": "e34f5897-e225-47f5-a4fb-13fa6099edc5",
        "playtest": false,
        "publicationDate": "2013-08-20",
        "publisherName": "Spelljammer Publishing",
        "slug": "crime-in-flumphtownville",
        "title": "Crime in Flumphtownville",
      },
      Object {
        "abbreviation": "sf",
        "id": "4a0db3d6-0477-4ef0-9584-7723ffe2efd3",
        "playtest": false,
        "publicationDate": "2013-11-19",
        "publisherName": "Spelljammer Publishing",
        "slug": "secrets-flumph",
        "title": "Secrets of the Flumph",
      },
    ]
  }
  message="There are no books matching the criteria."
/>
`;

exports[`<BooksTable /> with books: empty Array should match the snapshot 1`] = `
<Table
  cellProps={
    Object {
      "onDelete": [MockFunction],
    }
  }
  columns={
    Array [
      Object {
        "label": "Title",
        "prop": "title",
      },
      Object {
        "label": "Publisher",
        "prop": "publisherName",
      },
      Object {
        "label": " ",
        "prop": "actions",
        "value": " ",
      },
    ]
  }
  data={Array []}
  message="There are no books matching the criteria."
/>
`;
